src/components/CharactersPage/CharactersPage.js

```diff
@@ -48,7 +50,7 @@ export default function CharactersPage() {
         onTablePage={setTablePage}
         onRowClick={() => console.log("Table clicked")}
         onFetchMore={fetchMore}
-        onRefetch={refetch}
+        onFetchCharacters={fetchCharacters}
```

Comment:
Somehow the bug of data becomes undefined is related with `refetch` function. So to fix it, I used the execute function of the `useLazyQuery`